workerTypes/index.d.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Context } from '@azure/functions';
+
+/**
+ * This module is shipped as a built-in part of the Azure Functions Node.js worker and is available at runtime
+ */
+declare module '@azure/functions-worker' {
+    /**
+     * Register a hook to interact with the lifecycle of Azure Functions.
+     * Hooks are executed in the order they were registered and will block execution if they throw an error
+     */
+    export function registerHook(
+        hookName: 'preInvocation',
+        callback: (context: PreInvocationContext) => void | Promise<void>
+    ): Disposable;
+    export function registerHook(
+        hookName: 'postInvocation',
+        callback: (context: PostInvocationContext) => void | Promise<void>
+    ): Disposable;
+    export function registerHook(hookName: string, callback: HookCallback): Disposable;
+
+    export type HookCallback = (context: {}) => void | Promise<void>;
+
+    /**
+     * Context on a function that is about to be executed
+     * This object will be passed to all pre and post invocation hooks and can be used to share information between hooks
+     */
+    export interface PreInvocationContext {
+        /**
+         * The context object passed to the function
+         */
+        invocationContext: Context;
+
+        /**
+         * The input values for this specific invocation. Changes to this array _will_ affect the inputs passed to your function
+         */
+        inputs: any[];
+    }
+
+    /**
+     * Context on a function that has just executed
+     * This object will be passed to all pre and post invocation hooks and can be used to share information between hooks
+     */
+    export interface PostInvocationContext extends PreInvocationContext {
+        /**
+         * The result of the function, or null if there is no result. Changes to this value _will_ affect the overall result of the function
+         */
+        result: any;
+
+        /**
+         * The error for the function, or null if there is no error. Changes to this value _will_ affect the overall result of the function
+         */
+        error: any;
```

Comment:
I would support explicitly adding a `context` property here, to make it clear that the `context` object can be used to communicate between post and pre invocation hooks.

Perhaps also add the `input` property, since it also ends up being passed to the post invocation context?

Reply:
Ignore this, other comments have clarified my understanding and I know notice the `extends`